src/modules/rewards/Rewards.tsx

```diff
@@ -12,8 +14,20 @@ const Rewards: FC<RewardsProps> = () => {
   const [activeTab, setActiveTab] = useState(rewardsTabsList[0].value);
 
   const { isWalletConnected } = useAccount();
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  const locationArray = location.pathname.split('/');
+
+  const handleSetActiveTab = (tab: RewardsTabs) => {
+    setActiveTab(tab);
+    navigate(`/rewards/${tab}`);
+  };
 
-  const handleSetActiveTab = (tab: RewardsTabs) => setActiveTab(tab);
+  useEffect(() => {
+    if (locationArray.length === 3) setActiveTab(locationArray[2] as RewardsTabs);
+    else setActiveTab(rewardsTabsList[0].value);
+  }, [location]);
```

Comment:
This logic might needs some change since it doesn't redirects user to /dashboard from /rewards